go/libraries/doltcore/merge/violations_fk_prolly.go

```diff
@@ -169,31 +168,14 @@ func prollyChildSecDiffFkConstraintViolations(
 				}
 			}
 
-			if parentSecIdxCur == nil {
```

Comment:
Hit to merge perf is kind of painful. Can we hide this logic in a function in prolly to maintain the private cursor and get the rest of changes through?

Reply:
addressed this [098d40a](https://github.com/dolthub/dolt/pull/5533/commits/098d40a44f669dee5508b7a9309d30c686934da4) with `prolly.Map.HasPrefix()`